qiskit_experiments/framework/experiment_data.py

```diff
@@ -652,16 +655,18 @@ def _set_backend(self, new_backend: Backend, recursive: bool = True) -> None:
         provider = self._backend_data.provider
         if provider is not None:
             self._set_hgp_from_provider(provider)
+        # qiskit-ibm-runtime style
+        elif hasattr(self._backend, "_instance"):
+            try:
```

Comment:
This catches the `MagicMock` backend in tests which otherwise throws an error when trying to set the hgp.

Reply:
I am wary of changing the code to accommodate the tests. In the relevant tests, could we do something like

```python
job.backend.return_value = mock.create_autospec(Backend, instance=True)
```

`_instance` is not in the `Backend` base class so this will just skip this `if` clause here.

Reply:
That's a nice fix, thanks. Done in [c4fc480](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1371/commits/c4fc48072f995406dea5524305affbad603267d1).